tests/components/tautulli/test_config_flow.py

```diff
@@ -0,0 +1,227 @@
+"""Test Tautulli config flow."""
+from unittest.mock import AsyncMock, patch
+
+from pytautulli import exceptions
+from pytest import LogCaptureFixture
+
+from homeassistant import data_entry_flow
+from homeassistant.components.tautulli.const import DEFAULT_NAME, DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_REAUTH, SOURCE_USER
+from homeassistant.const import CONF_API_KEY, CONF_SOURCE
+from homeassistant.core import HomeAssistant
+
+from . import (
+    CONF_DATA,
+    CONF_IMPORT_DATA,
+    NAME,
+    patch_config_flow_tautulli,
+    setup_integration,
+)
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_flow_user_single_instance_allowed(hass: HomeAssistant) -> None:
+    """Test user step single instance allowed."""
+    entry = MockConfigEntry(domain=DOMAIN, data=CONF_DATA)
+    entry.add_to_hass(hass)
+
+    with patch_config_flow_tautulli(AsyncMock()):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=CONF_IMPORT_DATA,
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "single_instance_allowed"
+
+
+async def test_flow_user(hass: HomeAssistant) -> None:
+    """Test user initiated flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch_config_flow_tautulli(AsyncMock()):
+        result2 = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup function on all create entry results to avoid setting up the whole integration.